python/iceberg/api/transforms/projection_util.py

```diff
@@ -18,49 +18,138 @@
-            return Expressions.predicate(Operation.LT_EQ, name, transform.apply(boundary - 1))
-        elif pred.op == Operation.LT_EQ:
-            return Expressions.predicate(Operation.LT_EQ, name, transform.apply(boundary))
-        elif pred.op == Operation.GT:
-            return Expressions.predicate(Operation.GT_EQ, name, transform.apply(boundary + 1))
-        elif pred.op == Operation.GT_EQ:
-            return Expressions.predicate(Operation.GT_EQ, name, transform.apply(boundary))
-        elif pred.op == Operation.EQ:
-            return Expressions.predicate(pred.op, name, transform.apply(boundary))
-
-    def truncate_long(name, pred, transform):
-        return ProjectionUtil.truncate_integer(name, pred, transform)
-
-    def truncate_decimal(name, pred, transform):
-        boundary = pred.lit.value
-
-        if pred.op == Operation.LT:
-            minus_one = boundary - decimal.Decimal(1)
-            return Expressions.predicate(Operation.LT_EQ, name, transform.apply(minus_one))
-        elif pred.op == Operation.LT_EQ:
-            return Expressions.predicate(Operation.LT_EQ, name, transform.apply(boundary))
-        elif pred.op == Operation.GT:
-            plus_one = boundary + decimal.Decimal(1)
-            return Expressions.predicate(Operation.GT_EQ, name, transform.apply(plus_one))
-        elif pred.op == Operation.GT_EQ:
-            return Expressions.predicate(Operation.GT_EQ, name, transform.apply(boundary))
-        elif pred.op == Operation.EQ:
-            return Expressions.predicate(pred.op, name, transform.apply(boundary))
-
-    def truncate_array(name, pred, transform):
-        boundary = pred.lit.value
-
-        if pred.op == Operation.LT or pred.op == Operation.LT_EQ:
-            return Expressions.predicate(Operation.LT_EQ, name, transform.apply(boundary))
-        elif pred.op == Operation.GT or pred.op == Operation.GT_EQ:
-            return Expressions.predicate(Operation.GT_EQ, name, transform.apply(boundary))
-        elif pred.op == Operation.EQ:
-            return Expressions.predicate(pred.op, name, transform.apply(boundary))
+from ..expressions import Expressions, Operation
+from ..expressions.predicate import UnboundPredicate
+
+
+def truncate_integer(name, pred, transform):
+    boundary = pred.lit.value
+    if pred.op == Operation.LT:
+        return Expressions.predicate(Operation.LT_EQ, name, transform.apply(boundary - 1))
+    elif pred.op == Operation.LT_EQ:
+        return Expressions.predicate(Operation.LT_EQ, name, transform.apply(boundary))
+    elif pred.op == Operation.GT:
+        return Expressions.predicate(Operation.GT_EQ, name, transform.apply(boundary + 1))
+    elif pred.op == Operation.GT_EQ:
+        return Expressions.predicate(Operation.GT_EQ, name, transform.apply(boundary))
+    elif pred.op == Operation.EQ:
+        return Expressions.predicate(pred.op, name, transform.apply(boundary))
+
+
+def truncate_integer_strict(name, pred, transform):
```

Comment:
its not clear to me what the strict version is doing and why we need it? Its not being used anywhere in this patch. Maybe some unit tests would help?

Reply:
It has not been used in this patch. I added it as a placeholder so we track those not implemented functions in python.